File: app/src/main/java/com/returnnotfound/stain/base/BaseViewModel.java
package com.returnnotfound.stain.base;

import androidx.lifecycle.ViewModel;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public abstract class BaseViewModel extends ViewModel implements IBaseViewModel {
  private CompositeDisposable mCompositeDisposable = new CompositeDisposable();

  @Override
  public void addTask(Disposable disposable) {
    mCompositeDisposable.add(disposable);
  }

  @Override
  public void start() {

  }

  @Override
  public void onFirstResume() {

  }

  @Override
  public void onBackResume() {

  }

  @Override
  public void finish() {

  }

  @Override
  protected void onCleared() {
    mCompositeDisposable.dispose();
    super.onCleared();
  }
}

File: app/src/main/java/com/returnnotfound/stain/base/BaseViewModel.kt
package com.returnnotfound.stain.base

import androidx.lifecycle.ViewModel
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

abstract class BaseViewModel : ViewModel() {
  private val compositeDisposable: CompositeDisposable = CompositeDisposable()
  fun add(disposable: Disposable) {
    compositeDisposable.add(disposable)
  }

  override fun onCleared() {
    super.onCleared()
    compositeDisposable.dispose()
  }
}